### ISSUE TITLE
Returning non-JSON for a request

### ISSUE THREAD

**Issue by selewis** [SciGraph/SciGraph#246]
This URL returns "Failed to find class with ID: WB:WBGene00020993", which the JSON parser then barfs on: 
https://scigraph-data-dev.monarchinitiative.org/scigraph/graph/neighbors/WB:WBGene00020993.json?blankNodes=False&relationshipType=RO%3A0002205&entail=True&direction=OUTGOING&depth=1

The correct URL (below) returns JSON just fine. But whatever the insult this method should always return something in JSON format.
https://scigraph-data-dev.monarchinitiative.org/scigraph/graph/neighbors/WormBase:WBGene00020993.json?blankNodes=False&relationshipType=RO%3A0002205&entail=True&direction=OUTGOING&depth=1

**Comment by cmungall** [comment]
Yes, if response is 200, should return valid json. But there is an argument for returning 404 here.

@kltm @kshefchek @DoctorBud @tgbugs what is most consistent across our APIs - 404 or empty json?

**Comment by kltm** [comment]
I think it is best when possible/easy to return JSON when JSON is requested; separate from that is a 404, which should be returned when warranted--these should be considered separately.

**Comment by tgbugs** [comment]
For vocabulary services this already happens, see for example: http://neo.crbs.ucsd.edu:9000/scigraph/vocabulary/id/BIRNLEX:796 (json). This is inconsistent with http://neo.crbs.ucsd.edu:9000/scigraph/graph/BIRNLEX:796 (text). The 404 status should always be returned. Inconsistency with behavior between 1.5 and 2.0 (returning nothing for a 404, vs returning json) caused a bug in one of our consuming systems, it is much easier to avoid the issue in the future by returning a 404 status code so that we don't have to ever look at the content of the return statement.
